usr/lib/common/loadsave.c

```diff
@@ -811,7 +811,12 @@ CK_RV load_private_token_objects_old(STDLL_TokData_t *tokdata)
             fclose(fp2);
             continue;
         }
-        //size--;
+        if (size <= sizeof(CK_ULONG_32) + sizeof(CK_BBOOL)) {
+            fclose(fp2);
+            OCK_SYSLOG(LOG_ERR, "Improper size of object %s (ignoring it)\n",
+                       fname);
+            continue;
+        }
         size = size - sizeof(CK_ULONG_32) - sizeof(CK_BBOOL);
```

Comment:
It is not part of your change, but maybe you can also cover the next line as well

size -= sizeof(CK_ULONG_32) - sizeof(CK_BBOOL);

Reply:
I think I know why this was not done.  Your version is wrong!!!  The second minus then has to be a plus.

Reply:
You're right. It should be 
`size -= sizeof(CK_ULONG_32) + sizeof(CK_BBOOL);`